### PR TITLE
#710 Indexed blocker thread names JVM

### PR DESCRIPTION
**Pull request by Slakah**
Implemented #710

### CODE REVIEW

File: core/jvm/src/main/scala/cats/effect/internals/PoolUtils.scala [hunk 1]
```diff
@@ -34,8 +34,7 @@ private[internals] object PoolUtils {
       new ThreadFactory {
         val ctr = new AtomicInteger(0)
         def newThread(r: Runnable): Thread = {
-          val back = new Thread(r)
-          back.setName(s"ioapp-compute-${ctr.getAndIncrement()}")
+          val back = new Thread(r, s"ioapp-compute-${ctr.getAndIncrement()}")
```
Comment (Slakah) [inline core/jvm/src/main/scala/cats/effect/internals/PoolUtils.scala]: Inlined the thread name argument as per the `BlockerPlatform` implementation.

Can revert this change if requested.

File: core/jvm/src/main/scala/cats/effect/internals/BlockerPlatform.scala [hunk 2]
```diff
@@ -27,8 +28,9 @@ private[effect] trait BlockerPlatform {
    */
   def apply[F[_]](implicit F: Sync[F]): Resource[F, Blocker] =
     fromExecutorService(F.delay(Executors.newCachedThreadPool(new ThreadFactory {
+      val ctr = new AtomicInteger(0)
```
Comment (djspiewak) [inline core/jvm/src/main/scala/cats/effect/internals/BlockerPlatform.scala]: I actually wonder if it's worth putting the counter in a global location. The problem is that, unlike the global pool, people can create as many `Blocker` instances as they like. It's more conventional to only have one, but very much possible to have a ton of them.

File: core/jvm/src/main/scala/cats/effect/internals/PoolUtils.scala (hunk 1 above)
Comment (djspiewak) [reply core/jvm/src/main/scala/cats/effect/internals/PoolUtils.scala]: Seems better this way to me!

File: core/jvm/src/main/scala/cats/effect/internals/BlockerPlatform.scala (hunk 2 above)
Comment (Slakah) [reply core/jvm/src/main/scala/cats/effect/internals/BlockerPlatform.scala]: Yeah, very good spot. Will make the change now.

File: core/jvm/src/main/scala/cats/effect/internals/BlockerPlatform.scala [hunk 3]
```diff
@@ -19,16 +19,20 @@ package internals
 
 import scala.concurrent.ExecutionContext
 import java.util.concurrent.{ExecutorService, Executors, ThreadFactory}
+import java.util.concurrent.atomic.AtomicInteger
 import cats.data.NonEmptyList
 
 private[effect] trait BlockerPlatform {
+
+  private val threadCtr = new AtomicInteger(0)
```
Comment (Slakah) [inline core/jvm/src/main/scala/cats/effect/internals/BlockerPlatform.scala]: As per https://github.com/typelevel/cats-effect/pull/722#discussion_r355536484, this does somewhat assume that `BlockPlatform` will continue to only be extended by a single object `Blocker`.

The other option would be to have an object to hold the `AtomicInteger` reference, maybe:
```scala
private object BlockerPlatform {
  val threadCtr = new AtomicInteger(0)
}
```
It felt a touch too involved to do that.

Comment (rossabaker) [reply core/jvm/src/main/scala/cats/effect/internals/BlockerPlatform.scala]: Only one subtype is a safe assumption, and if it's wrong, duplicate thread names aren't tragic and can be fixed without binary breakage.
